### PR TITLE
[Backport] Remove datanode timetick mq, use rpc to report instead

### PR DESCRIPTION
**Pull request by wayblink**
#22953
master: #23156

### CODE REVIEW

**Comment by wayblink** [comment]
rerun ut

**Comment by wayblink** [comment]
rerun ut

**Comment by wayblink** [comment]
rerun ut

**Comment by wayblink** [comment]
rerun ut

**Comment by wayblink** [comment]
rerun ut

**Comment by wayblink** [comment]
rerun ut

File: internal/datanode/timetick_sender.go [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+	mu                  sync.Mutex
+	channelStatesCaches map[string]*segmentStatesSequence // string -> *segmentStatesSequence
+}
+
+// data struct only used in timeTickSender
+type segmentStatesSequence struct {
+	data map[uint64][]*datapb.SegmentStats // ts -> segmentStats
+}
+
+func newTimeTickManager(dataCoord types.DataCoord, nodeID int64) *timeTickSender {
```
Comment (czs007) [inline internal/datanode/timetick_sender.go:51]: better to renamed as newTimeTickSender

Comment (wayblink) [reply internal/datanode/timetick_sender.go]: fix it

File: internal/proto/data_coord.proto [hunk 2]
```diff
@@ -688,4 +690,9 @@ message GcConfirmRequest {
 message GcConfirmResponse {
   common.Status status = 1;
   bool gc_finished = 2;
-}
\ No newline at end of file
+}
+
+message ReportDataNodeTtMsgsRequest {
+  common.MsgBase base = 1;
```
Comment (czs007) [inline internal/proto/data_coord.proto:696]: Add a corresponding MsgType?

Comment (wayblink) [reply internal/proto/data_coord.proto:696]: Currently I use MsgType_DataNodeTt. ReportDataNodeTtMsgsRequest is a simple wrapper of DataNodeTt array. How about I add it in another PR? Put it to master first and pick to 2.2?

File: internal/datanode/timetick_sender.go [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+func (m *timeTickSender) update(channelName string, timestamp uint64, segmentStats []*datapb.SegmentStats) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	channelStates, ok := m.channelStatesCaches[channelName]
+	if !ok {
+		channelStates = &segmentStatesSequence{
+			data: make(map[uint64][]*datapb.SegmentStats, 0),
+		}
+	}
+	channelStates.data[timestamp] = segmentStats
```
Comment (czs007) [inline internal/datanode/timetick_sender.go:82]: channelStates.data[timestamp] = append(channelStates.data[timestamp], segmentStats)

Comment (wayblink) [reply internal/datanode/timetick_sender.go]: updated

**Comment by wayblink** [comment]
rerun ut

**Comment by wayblink** [comment]
rerun ut

**Comment by wayblink** [comment]
rerun ut

**Comment by yanliang567** [comment]
ci-e2e passed, but not submit the results. manual pass

**Pull request merged**